components/lib/datatable/DataTable.js

```diff
@@ -898,14 +898,14 @@ export const DataTable = React.forwardRef((props, ref) => {
         let value = [...data];
 
         if (columnSortable.current && columnSortFunction.current) {
-            value = columnSortFunction.current({ field, order });
+            value = columnSortFunction.current({ value, field, order });
```

Comment:
Hi @melloware,

I think you need to change `value` with `rowData: value`. Isn't it?